### PR TITLE
Travel

### PR DESCRIPTION
**Pull request by jackcarlisle**
## ready for review

* hooks up the travel info section to make a new query with new information 
* adds tests for the datepicker component taken from numo ui

### CODE REVIEW

File: lib/datepicker/Datepicker.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      monthClassName,
+      prevMonthClassName,
+      nextMonthClassName
+    } = this.props;
+
+    if (!this.state.isCalendarOpen) {
+      return null;
+    }
+
+    return <Calendar ref='calendar'
```
Comment (nikhilaravi) [inline lib/datepicker/Datepicker.js]: Can you fix the indentation.....?

File: lib/search-summary/index.js [hunk 2]
```diff
@@ -25,22 +25,75 @@ export default class SearchSummary extends Component {
...
+      numberOfAdultsTitle,
+      numberOfChildrenTitle,
+      durationTitle,
+      departureAirport,
+      duration
+    } = this.props;
     const stateStyle = this.state.active ? 'changeInputActive' : 'changeInputInactive';
     const blueContainerStyle = this.state.active ? 'blueContainerActive' : 'blueContainer';
     const header = this.state.active ? 'headerInactive' : 'headerActive';
     const adultOptions = [1, 2, 3, 4, 5, 6];
```
Comment (nikhilaravi) [inline lib/search-summary/index.js]: You could move these out into a separate options file to make this component a bit more readable

File: lib/search-summary/index.js [hunk 3]
```diff
@@ -51,14 +104,21 @@ export default class SearchSummary extends Component {
                 <h1 className='spiesTitle'>SPIES</h1>
                 <div className='inspirationTitle'>Inspiration</div>
               </div>
-              <DropDown width={'95%'} label={'FLYING FROM'} options={departureOptions} />
-              <DropDown width={'95%'} label={'DURATION'} options={durationOptions} />
+              <DropDown width={'95%'} label={'FLYING FROM'} setValue={setDepartureAirport} valueDefault={departureAirport} options={departureOptions} optionsTitle={'-'}/>
```
Comment (nikhilaravi) [inline lib/search-summary/index.js]: Can you separate out the props into separate lines for readability?

File: src/actions/search-results.js [hunk 4]
```diff
@@ -130,20 +134,128 @@ export function updateDisplayedItems (results) {
 }
 
 /**
+<<<<<<< HEAD
```
Comment (nikhilaravi) [inline src/actions/search-results.js]: left over from a merge conflict?

File: src/actions/search-results.js [hunk 5]
```diff
@@ -130,20 +134,128 @@ export function updateDisplayedItems (results) {
...
+}
+
+function combinePassengersForQuery (childAgeArray, numberOfChildren, numberOfAdults) {
+  const slicedChildAgeArray = childAgeArray.slice(0, Number(numberOfChildren));
+  const childPassengers = slicedChildAgeArray.map(slicedChildAge => {
+    const date = new Date();
+    const year = date.getFullYear() - Number(slicedChildAge.split(' ')[0]);
+    const month = ('0' + date.getMonth()).slice(-2);
+    const day = ('0' + date.getDate()).slice(-2);
+    return (
```
Comment (nikhilaravi) [inline src/actions/search-results.js]: you can just return the object on the same line there's no need to spread it out over three lines!

File: src/actions/search-results.js [hunk 6]
```diff
@@ -130,20 +134,128 @@ export function updateDisplayedItems (results) {
...
+    } = getState();
+    const childAgeArray = [childAge1, childAge2, childAge3, childAge4];
     if (tags.length > 0) {
       dispatch(busySearching());
-      const query = formatQuery(tags);
+      const formattedTags = formatQuery(tags);
+      const passengers = combinePassengersForQuery(childAgeArray, numberOfChildren, numberOfAdults);
+      const departureAirports = constructDepartureAirportQuery(departureAirport);
+      const travelPeriod = constructTravelPeriodQuery(departureDate, duration);
+      const query = {passengers: passengers, travelPeriod: travelPeriod, departureAirports: departureAirports, ...formattedTags};
```
Comment (nikhilaravi) [inline src/actions/search-results.js]: Spread the query object out over a few lines to make it more readable!

**Comment by nelsonic** [comment]
@nikhilaravi are you on the case to review/merge this or do you want to assign it to someone else? ❓

**Comment by nikhilaravi** [comment]
@nelsonic I reviewed some of it but if you have time to finish off the review today that would be really useful!

**Pull request merged**